compiler/luci/pass/src/TransformHardSwishToRelu6Pass.test.cpp

```diff
@@ -0,0 +1,146 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/TransformHardSwishToRelu6Pass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+/**
+ *  HardSwish graph
+ *
+ *        [CircleInput]
+ *              |
+ *              |
+ *      [CircleHardSwish]
+ *              |
+ *              |
+ *        [CircleOutput]
+ */
+struct HardSwishGraph
+{
+  loco::Graph _g;
+  luci::CircleInput *_input = nullptr;
+  luci::CircleHardSwish *_hardswish = nullptr;
+  luci::CircleOutput *_output = nullptr;
+};
+
+class TransformHardSwishToRelu6PassTest : public ::testing::Test
+{
+protected:
+  virtual void SetUp()
+  {
+    const int N = 1;
+    const int H = 4;
+    const int W = 4;
+    const int C = 3;
+
+    // graph input and output
+    auto graph_input = _hardswish_g._g.inputs()->create();
+    auto graph_output = _hardswish_g._g.outputs()->create();
+
+    // CircleInput
+    _hardswish_g._input = _hardswish_g._g.nodes()->create<luci::CircleInput>();
+    _hardswish_g._input->index(graph_input->index());
+    _hardswish_g._input->shape({N, H, W, C});
+    _hardswish_g._input->dtype(loco::DataType::FLOAT32);
+    _hardswish_g._input->name("input");
+
+    // CircleHardSwish
+    _hardswish_g._hardswish = _hardswish_g._g.nodes()->create<luci::CircleHardSwish>();
+    _hardswish_g._hardswish->features(_hardswish_g._input);
+    _hardswish_g._hardswish->shape({N, H, W, C});
+    _hardswish_g._hardswish->dtype(loco::DataType::FLOAT32);
+    _hardswish_g._hardswish->name("hardswish");
+
+    // CircleOutput
+    _hardswish_g._output = _hardswish_g._g.nodes()->create<luci::CircleOutput>();
+    _hardswish_g._output->index(graph_output->index());
+    _hardswish_g._output->from(_hardswish_g._hardswish);
+    _hardswish_g._output->shape({N, H, W, C});
+    _hardswish_g._output->dtype(loco::DataType::FLOAT32);
+    _hardswish_g._output->name("output");
+  }
+
+protected:
+  luci::TransformHardSwishToRelu6Pass _pass;
+  HardSwishGraph _hardswish_g;
+};
+
+} // namespace
+
+TEST_F(TransformHardSwishToRelu6PassTest, name)
+{
+  auto const name = _pass.name();
+  ASSERT_NE(nullptr, name);
+}
+
+/**
+ *  Decomposed graph looks like below.
+ *
+ *      [CircleInput]  [CircleConst]
+ *          |    \       /
+ *          |     \     /
+ *          |   [CircleAdd]
+ *          |        |
+ *          |        |
+ *          \  [CircleRelu6] [CircleConst]
+ *           \        \        /
+ *            \        \      /
+ *             \      [CircleMul]
+ *              \       /
+ *               \     /
+ *             [CircleMul]
+ *                  |
+ *                  |
+ *             [CircleOutput]
+ *
+ */
+TEST_F(TransformHardSwishToRelu6PassTest, simple_test)
```

Comment:
If decomposed, how about check expected node is not `CircleHardSwish` ?
and it is `CircleMul` ?